Add tests and convert to using mutation observer

JessChampion opened the pull request.

Commit:
Add tests and convert to using mutation observer

Firesphere commented on .circleci/config.yml:
```diff
@@ -58,4 +58,5 @@ jobs:
       - run: vendor/bin/phpunit --coverage-clover=coverage.xml -d memory_limit=512M tests/unit
       - run: vendor/bin/phpcs --standard=phpcs.xml.dist src tests/unit
+      - run: yarn test:coverage
       - run: bash <(curl -s https://codecov.io/bash) -f coverage.xml
```
It seems CodeCov doesn't really support jest :'(

Could you add the artifacts to circleci? It's at least something

Commit:
Added the else-flow tests

Also, found a bug due to testing, length of ['start'] could be different
if the input is not an array but a single value.

Signed-off-by: Jess Champion <jess@silverstripe.com>

Firesphere commented:
Thanks for the refactor and tests! 🙇
